ext/cl_khr_layers.asciidoc

```diff
@@ -0,0 +1,183 @@
+    _param_value_.
+    This size in bytes must be ≥ to the size of return type specified in the
+    <<layer-queries-table, Layer Queries>> table.
+  * _param_value_ is a pointer to memory location where appropriate values for a
+    given _param_name_, as specified in the <<layer-queries-table, Layer
+    Queries>> table, will be returned.
+    If _param_value_ is `NULL`, it is ignored.
+  * _param_value_size_ret_ returns the actual size in bytes of data being
+    queried by _param_name_.
+    If _param_value_size_ret_ is `NULL`, it is ignored.
+
+The information that can be queried using *clGetLayerInfo* is specified
+in the <<layer-queries-table, Layer Queries>> table.
+
+[[layer-queries-table]]
+.List of supported param_names by <<clGetLayerInfo>>
+[width="100%",cols="<34%,<33%,<33%",options="header"]
+|====
+| Layer Info | Return Type | Description
+| *CL_LAYER_API_VERSION* | cl_layer_api_version | OpenCL Layer API version
+|====
+
+*clGetLayerInfo* returns *CL_SUCCESS* if the function is executed
+successfully.
+Otherwise, it returns one of the following errors.
+
+  * *CL_INVALID_VALUE* if _param_name_ is not one of the supported values or
+    if size in bytes specified by _param_value_size_ is < size of return
+    type as specified in the <<layer-queries-table, Layer Queries>> table,
+    and _param_value_ is not a `NULL` value.
+
+==== Layer Initialization
+
+[open,refpage='clInitLayer',desc='Initialize an OpenCL layer',type='protos']
+Initialization of a Layer can be achieved with the function:
+[source,c]
+
+----
+cl_int clInitLayer(cl_uint                         num_entries,
+                   const struct _cl_icd_dispatch  *target_dispatch,
+                   cl_uint                        *num_entries_ret,
+                   const struct _cl_icd_dispatch **layer_dispatch_ret);
+----
+
+  * _num_entries_ is the number of entry provided by the Loader to
+    the layer.
+  * _target_dispatch_ is the dispatch table, provided by the loader, that
+    the layer must redirect it's call to.
+  * _num_entries_ret_ returns the number of entries in the dispatch table the
+    layer will be returning.
+  * _layer_dispatch_ret_ return the address of the layer dispatch table. The
+    table can contain `NULL` value entries, and the loader is responsible for
+    providing the missing entries using the entries from the target dispatch.
+
+*clInitLayer* returns *CL_SUCCESS* if the function is executed
+successfully.
+Otherwise, it returns one of the following errors.
+
+  * *CL_INVALID_VALUE* if _num_entries_ is sufficient for the layer or if
+    _target_dispatch_ is a `NULL` value, or _num_entries_ret_ is a `NULL`
+    value, or _layer_dispatch_ret_ is a `NULL` value.
+
+[[cl_khr_layer-source-code]]
+=== Source Code
+
+The official source for the ICD loader is available on github, at:
+
+https://github.com/KhronosGroup/OpenCL-ICD-Loader
+The header file *include/cl_icd_layer.h* defines the necessary types
+and API entry points.
```

Comment:
I think we'll want to eventually include the layer APIs and defines in the official API headers.  Do you think this makes sense, or should we be sign-posting to the ICD loader headers instead?

My first reaction is to include them in the extension headers.  I think that'd make sense, since this is a "loader extension".  I'm open to other alternatives though.

Reply:
> I think we'll want to eventually include the layer APIs and defines in the official API headers. Do you think this makes sense, or should we be sign-posting to the ICD loader headers instead?
> 
> My first reaction is to include them in the extension headers. I think that'd make sense, since this is a "loader extension". I'm open to other alternatives though.

see above